T02/Program.cs

```diff
@@ -0,0 +1,138 @@
+﻿namespace T02;
+using System.Text;
+using static State;
+using static System.Console;
+
+class Program {
+   static void Main () {
+      Random rand = new ();
+      for (int i = 0; i < 100; i++) ToWordAndRoman (rand.Next (1, 4000));
+      while (true) {
+         Write ("Enter the number : ");
+         if (int.TryParse (ReadLine (), out int n)) {
+            if (n <= 3999) ToWordAndRoman (n);
+            else WriteLine ("Max Limit is 3999\n");
+         } else { WriteLine ("Conversion Failed"); break; }
+      }
+   }
+
+   static (string word, string roman) Convert (int n) {
+      var (tmp, prev, last) = (n, 0, 0);
+      Action none = () => { }, todo;
+      State s = A;
+      StringBuilder word = new (), roman = new ();
+      while (tmp > 0) {
+         var rem = tmp % 10; // Remainder
+         tmp /= 10;
+         var (remWord, remRoman) = (words[rem], romans[rem]);
+         bool remZero = rem is 0;
+         var val = int.Parse (rem is 1 ? $"{rem}{prev}" : $"{rem}0");
+         (s, todo) = (s, remZero) switch {
+            (A, true) => (B, none),
+            (A, false) => (B, () => { word.Insert (0, remWord); roman.Append (remRoman); prev = rem; last = rem; }),
+            (B, true) => (C, none),
+            (B, false) => (C, () => {
+               if (rem is 1) word.Clear (); word.Insert (0, words[val] + " ");
+               switch (val) {
+                  case 40 or 90: roman.Insert (0, romans[val]); break;
+                  case >= 50 and < 90:
+                     for (int j = 0, k = (val - 50) / 10; j < k; j++) roman.Insert (0, "X"); roman.Insert (0, "L"); break;
+                  default: for (int j = 0, k = val / 10; j < k; j++) roman.Insert (0, "X"); break;
+               }
+               last = rem;

```

Comment:
```suggestion
               last = rem;
               }),
```
The braces and comma could have been in same line for each case